crates/flux-refineck/src/checker.rs

```diff
@@ -1139,6 +1168,14 @@ impl<'ck, 'genv, 'tcx, M: Mode> Checker<'ck, 'genv, 'tcx, M> {
         Ok(ty)
     }
 
+    fn discr_to_int_cast(adt_def: &AdtDef, bty: BaseTy) -> Ty {
+        let vals = adt_def
+            .discriminants()
+            .map(|(_, idx)| Expr::eq(Expr::nu(), Expr::from_bits(&bty, idx)))
+            .collect_vec();
```

Comment:
This could be a huge list. We can optimize it by finding contiguous intervals. Presumably, that could be easier for z3, but it's fine if we don't. I doubt this will cause problems now, but it sounds like a fun thing to implement!

Reply:
Yes, I'll leave a TODO there for now, not clear what the best approach is... its easy and more efficient but "less precise" to make it an interval..